File: src/main/java/com/sedmelluq/discord/lavaplayer/player/event/AudioEvent.java
package com.sedmelluq.discord.lavaplayer.player.event;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;

/**
 * An event related to an audio player.
 */
public abstract class AudioEvent {
  /**
   * The related audio player.
   */
  public final AudioPlayer player;

  /**
   * @param player The related audio player.
   */
  public AudioEvent(AudioPlayer player) {
    this.player = player;
  }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/manager/event/AudioEvent.kt
package com.sedmelluq.discord.lavaplayer.manager.event

import com.sedmelluq.discord.lavaplayer.manager.AudioPlayer

/**
 * An event related to an audio player.
 */
interface AudioEvent {
    /**
     * The related audio player.
     */
    val player: AudioPlayer
}
